### DISCUSSION TITLE
Why does queries.getByLabelText give me an error with the internal code while trying to run Lighthouse user flows

### DISCUSSION THREAD

**Discussion by aadisarma7** [GoogleChrome/lighthouse#15606] [Q&A] [answered]
I am trying to replicate the example code from the lighthouse documentation (https://github.com/GoogleChrome/lighthouse/blob/main/docs/user-flows.md) The code block I am interested in is complete user flow code.

import {writeFileSync} from 'fs';
import puppeteer from 'puppeteer';
import * as pptrTestingLibrary from 'pptr-testing-library';
import {startFlow} from 'lighthouse';

const {getDocument, queries} = pptrTestingLibrary;

async function search(page) {
  const $document = await getDocument(page);
  const $searchBox = await queries.getByLabelText($document, /type to search/i);
  await $searchBox.type('Xbox Series X');
  await Promise.all([
    $searchBox.press('Enter'),
    page.waitForNavigation({waitUntil: ['load', 'networkidle2']}),
  ]);
}

// Setup the browser and Lighthouse.
const browser = await puppeteer.launch();
const page = await browser.newPage();
const flow = await startFlow(page);

// Phase 1 - Navigate to the landing page.
await flow.navigate('https://www.bestbuy.com');

// Phase 2 - Interact with the page and submit the search form.
await flow.startTimespan();
await search(page);
await flow.endTimespan();

// Phase 3 - Analyze the new state.
await flow.snapshot();

// Phase 4 - Navigate to a detail page.
await flow.navigate(async () => {
  const $document = await getDocument(page);
  const $link = await queries.getByText($document, /Xbox Series X 1TB Console/);
  $link.click();
});

// Get the comprehensive flow report.
writeFileSync('report.html', await flow.generateReport());
// Save results as JSON.
writeFileSync('flow-result.json', JSON.stringify(await flow.createFlowResult(), null, 2));

// Cleanup.
await browser.close();
This gives me the following error: 
node:internal/process/esm_loader:108 internalBinding('errors').triggerUncaughtException( ^

TypeError: containerHandle.executionContext is not a function

I have tested my code and found that the issue is on the lines where I run queries.getByLabelText and queries.getByText. However, I am not sure how to proceed from here. I am using lighthouse: 11.3.0, puppeteer: 21.5.1, and pptr-testing-library: 0.7.0. My package.json file is listed below in case it is relevant.

{
  "type": "module",
  "devDependencies": {
    "lighthouse": "^11.3.0",
    "pptr-testing-library": "^0.7.0"
  },
  "dependencies": {
    "puppeteer": "^21.5.1"
  }
}
I have also tried using other methods such as getByPlaceholderText to get the specified textbox from the webpage, but these give me the same error.

**Comment by adamraine** [comment, marked as answer]
Looks like `pptr-testing-library` is not compatible with the latest version of puppeteer https://github.com/GoogleChrome/lighthouse/issues/15607. We should update our docs, but you can still create a user flow without `pptr-testing-library`.

**Reply by ChristopherPHolder** [reply]
I am not sure if that is the issue with this user-flow. 

If you run it with headless false it will navigate to this page:

![image](https://github.com/GoogleChrome/lighthouse/assets/40126819/449bcb5d-e18a-484c-90fb-da6628ef61ea)

I am running the test from Austria. 

It makes sense that it crashes, because their is no search bar. 

I agree with removing `pptr-testing-library` from the docs, and doing the same thing that the test does with only puppeteer is probably cleaner and easer to understand. 

Will continue the conversation in the related issue #15607
